lint_diffs/__init__.py

```diff
@@ -266,55 +278,67 @@ def main():
 
     args = _parse_args()
 
+    # Read config files and override them with command line arguments
     py_config = read_config()
-
     _alter_config_with_args(args, py_config)
 
+    # Create actual config dictionary to be used ahead, saving some common
+    # settings in local variables as well.
     config = _config_to_dict(py_config)
-
+    strict = config.get("strict", False)
+    parallel = config.get("parallel", 1)
+    bare = config.get("bare", None)
     if config["debug"]:
         log.setLevel(logging.DEBUG)
 
     diffs = read_diffs()
-
     log.debug("diffs: %s", list(diffs))
 
-    linters = {}
+    linters = defaultdict(set)
     for fname in diffs:
         _, ext = os.path.splitext(fname)
         if ext in config:
             for linter_name in config[ext]:
-                if linter_name not in linters:
-                    linters[linter_name] = set()
                 linters[linter_name].add(fname)
+    log.debug("linters: %s, strict: %s, parallel: %s", linters, strict, parallel)
 
-    if not linters:
+    #
+    # BAIL OUT!
+    #
+    # Don't even try going ahead as we have no linter to be executed, or at
+    # least no linter to for current change-set extensions
+    #
+    if not linters or bare and bare not in linters:
         log.debug("no files need linting")
         return
 
-    strict = config.get("strict", False)
-    parallel = config.get("parallel", 1)
-
-    log.debug("linters: %s, strict: %s, parallel: %s", linters, strict, parallel)
-
     def print_lint(item):
         """Run linter, print output to screen."""
         linter, files = item
         ret = do_lint(config, linter, diffs, list(files))
         with CONSOLE_LOCK:
             print(ret.output)
-            print("=== %s: mine=%s, always=%s\n" % (linter, ret.mine, ret.always))
+            if not bare:
+                print("=== %s: mine=%s, always=%s\n" % (linter, ret.mine, ret.always))
         if ret.returncode == NOTFOUND:
             return 1 if strict else 0
         return ret.linted > 0 and (ret.returncode or 1)
 
-    if parallel > 1:
-        pool = multiprocessing.dummy.Pool(parallel)
-        mapper = pool.map
+    # Normal non-bare execution will run all linters we know about,
+    # optionally in parallel according current configuration
+    if bare is None:
+        if parallel > 1:
+            log.debug("Going parallel mode with %r", parallel)
+            pool = multiprocessing.dummy.Pool(parallel)
+            mapper = pool.map
+        else:
+            log.debug("Going ordinary model")
+            mapper = map
+        exitcode = max(mapper(print_lint, linters.items()))
```

Comment:
is this necessary?  exit code should be the same... bare should just force parallel==1

Reply:
That sounds like a better option indeed.

Reply:
I reduced code paths filtering linters dictionary before. That should help.